ticketing-eventing/mail-handler-service/src/main/java/de/muenchen/oss/dbs/ticketing/eventing/mailhandler/adapter/out/mail/TempFileInputStreamDataSource.java

```diff
@@ -0,0 +1,54 @@
+package de.muenchen.oss.dbs.ticketing.eventing.mailhandler.adapter.out.mail;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import jakarta.activation.DataSource;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * DataSource from InputStream which stores InputStream into tmp file for multiple reads.
+ */
+@Slf4j
+public class TempFileInputStreamDataSource implements DataSource {
+    private final String name;
+    private final String contentType;
+    @Getter
+    private final File tempFile;
+
+    @SuppressFBWarnings("CT_CONSTRUCTOR_THROW")
+    public TempFileInputStreamDataSource(final String name, final String contentType, final InputStream inputStream) throws IOException {
+        this.name = name;
+        this.contentType = contentType;
+        final File tmpFile = Files.createTempFile("attachment", null).toFile();
```

Comment:
This will always create a file named `attachment`. 
- If there are more than one attachment, will this still work? I.e. is one attachment handled sequentially after another and  the objects of type `TempFileInputStreamDataSource` do not need to exist next to each other?
- If there are more mail-handler instances, will they not potentially overwrite the same file?